controllers/userController.js

```diff
@@ -1,90 +1,214 @@
+      else resolve(res);
     })
-    .catch(error => res.status(400).send({message: message, error: error}));
+  });
+}
+
+//load with csv file
+exports.loadfruserLoadfromcsvomcsv = async (req, res) => {
+  const { file } = req.body;
+
+  // Fetching the data from each row 
+  // and inserting to the table "sample"
+  for (var i = 0; i < file.length; i++) {
+    var email = file[i]["email"],
+      name = file[i]["name"],
+      type = file[i]["type"],
+      visions = file[i]["visions"];
+
+    try {
+      let verify = await this.verifyUser( email );
+      if (verify.NUM > 0) {
+        return res.send({ status: STATUS_CODE.EMAIL_USED, message: email });
+      }
+      await this.insertUser({ email, name, type, visions });
+    } catch (error) {
+      return res.send({ status: STATUS_CODE.ERROR, message: error });
+    }
+  }
+
+  return res.send({ status: STATUS_CODE.SUCCESS });
 };
 
 //add one user
-exports.adduser = async (req, res) => {
-  const { email, name, type, visions} = req.body;
-  let message = 'Error in creating user';
-  const query = `INSERT INTO users 
-  (email, name, type, status, visions)
-  VALUES (?,?,?,'unregistered',?)`;
-
-  connection.promise().query(query, [email, name, type, visions])
-  .then(data => {
-    if (data[0].affectedRows) {
-      message = 'user created successfully';
-      res.status(200).send({message: message, data: data[0]});
+exports.createUser = async (req, res) => {
+  const { email, name, type, visions } = req.body;
+
+  try {
+    let verify = await this.verifyUser( email );
+    if (verify.NUM > 0) {
+      return res.send({ status: STATUS_CODE.EMAIL_USED, message: email });
+    }
+
+    let result = await this.insertUser({ email, name, type, visions });
+    
+    if (result.affectedRows) {
+      return res.send({ status: STATUS_CODE.SUCCESS });
+    }
+  } catch (error) {
+    return res.send({ status: STATUS_CODE.ERROR, message: error });
+  }
+
+  return res.send({ status: STATUS_CODE.ERROR });
+};
+
+//get all first year students, return a json object
+exports.readUser = async (req, res) => {
+  const name_sort = (!req.query.name_sort) ? ' name ASC' : ' name ' + req.query.name_sort;
+  const name_search = (!req.query.name_search) ? '' : ' name = ' + req.query.name_search;
+  const email_sort = (!req.query.email_sort) ? '' : ' email ' + req.query.email_sort;
+  const email_search = (!req.query.email_search) ? '' : ' email = ' + req.query.email_search;
+  const visions_sort = (!req.query.visions_sort) ? '' : ' visions ' + req.query.visions_sort;
+  const visions_filter = (!req.query.visions_filter) ? '' : ' visions = ' + req.query.visions_filter;
+  const status_filter = (!req.query.status_filter) ? '' : ' status = ' + req.query.status_filter;
+  const type_filter = (!req.query.type_filter) ? '' : ' type = ' + req.query.type_filter;
+  const row_start = (!req.query.row_start) ? 0 : req.query.row_start;
+  const row_num = (!req.query.row_end) ? 10 : req.query.row_end;
```

Comment:
There might be a typo. row_end should be row_num I guess:)

Reply:
Yea you are totally right